inc/toolbox.class.php

```diff
@@ -785,16 +785,15 @@ static function stripslashes_deep($value) {
    **/
    static function append_params($array, $separator='&', $parent='') {
 
-      $params = array();
+        $params = array();
       foreach ($array as $k => $v) {
 
          if (is_array($v)) {
             $params[] = self::append_params($v, $separator,
                                             (empty($parent) ? rawurlencode($k)
-                                                            : $parent .'['.rawurlencode($k).']'));
+                                                            : $parent . '%5B' . rawurlencode($k) . '%5D'));
```

Comment:
I get an W3C error on bad URL formating.
I find it on the ticket view.
The button ![image](https://cloud.githubusercontent.com/assets/6187125/21567639/edbb4caa-cead-11e6-850f-a30a68959cbe.png) have the search parameters on the href of the a element.

Without this patch, we provide an element like 
```
<a href="/glpi/front/ticket.php188?reset=reset&amp;criteria[0][field]=55&amp;criteria[0][searchtype]=equals&amp;criteria[0][value]=2&amp;criteria[0][link]=AND&amp;criteria[1][field]=59&amp;criteria[1][searchtype]=equals&amp;criteria[1][value]=2&amp;criteria[1][link]=AND"><img title="Ticket attendant votre validation" alt="Ticket attendant votre validation" src="/glpi/pics/menu_showall.png" class="pointer"></a>
```
This code will throw an error on W3C validator as "[" and "]" elements aren't allowed in a URL

With this patch, we provide an element like 
```
<a href="/glpi/front/ticket.php188?reset=reset&amp;criteria%5B0%5D%5Bfield%5D=55&amp;criteria%5B0%5D%5Bsearchtype%5D=equals&amp;criteria%5B0%5D%5Bvalue%5D=2&amp;criteria%5B0%5D%5Blink%5D=AND&amp;criteria%5B1%5D%5Bfield%5D=59&amp;criteria%5B1%5D%5Bsearchtype%5D=equals&amp;criteria%5B1%5D%5Bvalue%5D=2&amp;criteria%5B1%5D%5Blink%5D=AND"><img title="Ticket attendant votre validation" alt="Ticket attendant votre validation" src="/glpi/pics/menu_showall.png" class="pointer"></a>
```

(same for above)